rules/community/cloudtrail/cloudtrail_mfa_policy_abuse_attempt.py

```diff
@@ -0,0 +1,75 @@
+"""Alert on calls made without MFA that may be attempting to abuse a flawed enforcement policy"""
+from stream_alert.rule_processor.rules_engine import StreamRules
+
+rule = StreamRules.rule
+disable = StreamRules.disable()
+
+
+@rule(logs=['cloudtrail:events'],
+      matchers=[],
+      outputs=['aws-s3:sample-bucket',
+               'pagerduty:sample-integration',
+               'slack:sample-channel'])
+def cloudtrail_mfa_policy_abuse_attempt(rec):
+    """
+    author:           Scott Piper of Summit Route in collaboration with Duo Security
+    description:      Alert on potential attacks performed by users that are supposed to have
+                        MFA enforced. May indicate leaked access keys and an attempt to abuse
+                        a flawed MFA enforcement policy.
+    playbook:         (a) Identify whether or not the attempt was a mistake.
+                      (b) Begin IR: Roll access keys, investigate past CloudTrail logs for
+                      other actions performed, investigate how the keys were leaked in the
+                      first place.
+    """
+
+    # Depending on the practices you follow with AWS today, you may wish to simply alert
+    # on any errors at all in AWS, possibly with some focus on IAM actions,
+    # and possibly with some exemptions to ignore things like failed login attempts.
+    # Another option would be to just alert on any AccessDenied's when MFA is not used.
+    # This rule attempts to reduce the false positives.
+
+    # Get the value for whether the user is MFA authenticated.
+    # If this doesn't exist, assume not MFA authenticated.
+    mfa_authenticated = "false"
+    try:
+        mfa_authenticated = rec['userIdentity']['sessionContext']['attributes']['mfaAuthenticated']
+    except KeyError:
+        pass
```

Comment:
for brevity, the `mfa_authenticated = "false"` could be moved to within this `except` and the `pass` removed.

You could also string a bunch of `.gets` together here with a default value, but I don't think I love that approach:

`mfa_authenticated = rec['userIdentity'].get('sessionContext', {}).get('attributes', {}).get('mfaAuthenticated', 'false')`